public/app/features/dashboard-scene/panel-edit/PanelOptions.tsx

```diff
@@ -15,12 +15,15 @@ interface Props {
 }
 
 export const PanelOptions = React.memo<Props>(({ vizManager, searchQuery, listMode }) => {
-  const { panel } = vizManager.useState();
+  const { panel, repeat } = vizManager.useState();
   const { data } = sceneGraph.getData(panel).useState();
   const { options, fieldConfig } = panel.useState();
 
   // eslint-disable-next-line react-hooks/exhaustive-deps
-  const panelFrameOptions = useMemo(() => getPanelFrameCategory2(vizManager), [vizManager, panel]);
+  const panelFrameOptions = useMemo(
+    () => getPanelFrameCategory2(vizManager, panel, repeat),
+    [vizManager, panel, repeat]
+  );
```

Comment:
Why do we need this change?

Reply:
This is one spot I'm having doubts myself, so if there is a better way to do it I'm open to change it. 

We need these state values to send them later on to `RepeatRowSelect2` and with this refactor we make `RepeatRowSelect2` reusable for row controls as well. Before this, `RepeatRowSelect2` took the `VizPanelManager` and subscribed to state inside, but this cannot be used for row controls in the dashboard scene cuz there is no `VizPanelManager` involved.